### DISCUSSION TITLE
OxyPlot geospatial plotting in C# possible?

### DISCUSSION THREAD

**Discussion by dimales** [oxyplot/oxyplot#1874] [Q&A] [answered]
I'm coding in C# and looking for a way to plot the following:
- static (not necessarily interactive) world map in any of the projections (polar, for example);
- land surface outline on the map (just a shape, only to discern sea and land);
- some geospatial information (geographical points, isolines, etc).

Is it actually possible to do with OxyPlot?

(Before, I've used Python and it's matplotlib+basemap libraries to do the same, it was very easy and fulfilling. I'm looking for a way to do the same using C# solely).

**Comment by VisualMelon** [comment, marked as answer]
This sort of thing?

![image](https://user-images.githubusercontent.com/5646475/161534509-4173a621-5ac5-4f88-8f3c-2ac6794a9c94.png)

This is a series of `PolygonAnnotations` showing the data in https://github.com/balakrishnanc/gnugeoplot/blob/master/world-50m.txt (though I don't know what licencing concerns there would be with actually using this for non-educational purposes). The only problem is that OxyPlot currently imposes a limit on the resolution of Polygon annotations. This was resolved in https://github.com/oxyplot/oxyplot/pull/1854 but isn't published yet (that figure was made with a locally built copy of the library).

There is also an example in the example library showing how you can use e.g. open-street-map tiles to create an interactive map. Again, it implements this with annotations, and you can plot whatever series you like on top of them.

**Reply by dimales** [reply]
Yes, kinda like that, thank you! For now I'm trying to find out how to install OxyPlot (it's not so easy as in Python, doing it for the first time).
Could you please tell me: could I alter colours in such a plot (for example, you presented blue land, white sea, I would like it to be vice versa).
Could the land countour be bold and dark? And have more details (rivers, lakes, etc)?
Could I use my own basemap, and which formats? (see pic.) 
![image](https://user-images.githubusercontent.com/44398125/161585098-cb20841f-542a-4e45-8bfa-4cf03b790d60.png)

And finally, could you please give me a small little tip on how to actually draw such plots? For now I'm trying to explore how to view examples there, but still don't quite understand.

**Reply by VisualMelon** [reply]
Re. colours: yes, you can make the background blue and the polygons white; other features depends on what geospatial information you have, but you can have outlines etc

![image](https://user-images.githubusercontent.com/5646475/161590765-2ffeb358-7254-4ac5-b523-024cde0daeb0.png)

If you have lots of separate and small questions, perhaps ask in the gitter: it's a bit easier for free-form questions: https://gitter.im/oxyplot/oxyplot - I don't want to fill this with general C# information, but happy to have a to-fro in the chat if you are just starting out (depends on many things that aren't really OxyPlot specific). That map is actually a Powershell script I threw together for something a few months back, but I can port it to C# quickly if it would be helpful for you.

**Reply by dimales** [reply]
Yeah, I would really appreciate if you could port that to C#. Sorry, I'm not even close to be any proficient in C# or coding generally, but trying to study by doing something useful for me. And I'm confused by all the OOP, complexity of the libraries and the lack of information on how to use them.

**Reply by VisualMelon** [reply]
If you have shapefiles you want to plot, then you'd be better off using a proper GIS offering (e.g. ESRI/dotspatial/Mapsui); they'll be much better suited to this task. OxyPlot is a simple and portable graph plotting library and won't do any of the file handling for you as would a proper GIS offering.

This is the map example in C# in any case: https://gist.github.com/VisualMelon/24ff3c86482d7dc6d3af62078da17c9f

**Reply by dimales** [reply]
Thank you!
I'm looking forward to get rid of all the side applications, and fulfill everything with using C# and its libraries only - like it can be done with Python+matplotlib. Might you know any other C# library, that would offer the same tools as matplotlib does? 
Using some side GIS offerings is admissible for small apps/simple scripts, but ultimately, when we are talking about more or less complex application, it is required that everything could be done within the app only.

I've managed to build the same map you have. For some reason, I had to change all the "." to "," in world-50m.txt. Does it handle georeferencing? Like, if I need to put something to a plot by coordinates?
If I want to use other basemap shapefile (with conincal projection, for example), I have to somehow transform shapefile to a similar .txt with coordinates?

**Reply by VisualMelon** [reply]
There is SciSharp (which provides bindings either for the python APIs, or maybe the underlying C APIs for the key Python numerical libs), but I've never used it: https://github.com/SciSharp/Matplotlib.Net

I suspect you are unlikely to find a similar home-grown library: MatPlotLib has a very MATLAB-esq API, and that is not natural in .NET.

**Reply by dimales** [reply]
Ok, thanks, I got it!
